### ISSUE TITLE
Change default timezone

### ISSUE THREAD

**Issue by juliardi** [codemix/yii2-excelexport#14]
I am trying to display data in datetime format. But, PHPExcel convert it by default to UTC timezone. Is there any way to change the default timezone through your library?

**Comment by mikehaertl** [comment]
There's no built in support for timezone conversion but you can use the `formatter` option to do define a callback where you can convert your values to correct the timezone. Have a look at the raw data example: https://github.com/codemix/yii2-excelexport#raw-data.

Here's an example how I did timezone conversion in one of my projects. Note though, that this is using PHPExcel directly without my library here. You should be able to use something like this in your `formatter` callback:

```php
        // PHPExcel is lacking correct timezone support.
        // The PHPExcel_Shared_Date::PHPToExcel() method expects a "pseudo-timestamp":
        // Something like a UNIX timestamp but including local timezone offset.
        // The offset from UTC to Europe/Berlin is not constant but differs
        // between 3600 s in winter and 7200 s in summer times to due DST.
        // We use PHP's DateTimeZone class to calculate the correct offset for each date.
        $timeZoneUtc = new DateTimeZone('UTC');
        $timeZoneEurope = new DateTimeZone('Europe/Berlin');

        foreach ($this->dataProvider->models as $price) {

            // The DateTime object in UTC timezone (DB time)
            $timeUtc = new \Datetime($price->start_from, $timeZoneUtc);

            // The current offset for that date to the Europe/Berlin timezone (3600 or 7200 s)
            $offsetEurope = $timeZoneEurope->getOffset($timeUtc);

            // The manually calculated excel timestamp as UTC + current offset
            $excelTime = $timeUtc->getTimestamp() + $offsetEurope;
            $sheet->setCellValue('A'.$row, PHPExcel_Shared_Date::PHPToExcel($excelTime));

            $sheet->getStyle('A'.$row)
                ->getNumberFormat()
                ->setFormatCode('dd.mm.yyyy hh:mm');
```

**Comment by juliardi** [comment]
Thank you for your answer. Previously, I solved this by extending `codemix\excelexport\ActiveExcelSheet` class and override the `getFormatters` metho d.
But I think I will use the formatter option like this :
```php
'formatters' => [
    9 => function($v) {
        $timestamp = strtotime($v);
        $year = date('Y', $timestamp);
        $month = date('m', $timestamp);
        $day = date('d', $timestamp);
        $hour = date('H', $timestamp);
        $minute = date('i', $timestamp);
        $second = date('s', $timestamp);
        return \PHPExcel_Shared_Date::FormattedPHPToExcel($year, $month, $day,
                                                        $hour, $minute, $second);

    },
],
```
I don't quite understand the difference between `PHPExcel_Shared_Date::FormattedPHPToExcel()` and `PHPExcel_Shared_Date::PHPToExcel()`, but the former works for me

**Comment by mikehaertl** [comment]
I'm not really sure either, but you can easily check the sources. `PHPToExcel()` seems to call `FormattedPHPToExcel()` internally. There also is a parameter `$adjustToTimezone` in `PHPToExcel()` but it's not used at all. I'd say, this is a bug in PHPOffice - or at least an incomplete feature.

So if things work now for you I'm closing this.

**Comment by juliardi** [comment]
I noticed that they implemented `$adjustToTimezone` in phpexcel repository branch 1.8, but somehow it's not included in their 1.8.0 release. Also I think it's better to switch to phpspreadsheet once it's released.

**Comment by mikehaertl** [comment]
> Also I think it's better to switch to phpspreadsheet once it's released.

Oh, that's looks cool, didn't know this project. Will keep an eye on it.

**Comment by mikehaertl** [comment]
@juliardi I've just released 2.3.1 which includes automatic timezone conversion in ActiveExcelSheet:

> **Note** Since version 2.3.1 datetime attributes will automatically be  converted to the correct timezone. This feature makes use of the current  [defaultTimeZone](http://www.yiiframework.com/doc-2.0/yii-i18n-formatter.html#$defaultTimeZone-detail) and [timeZone](http://www.yiiframework.com/doc-2.0/yii-base-application.html#getTimeZone()-detail) setting of the app.

**Comment by juliardi** [comment]
Okay, thank you for your notification
